src/crawlers/cheerio_crawler.js

```diff
@@ -274,10 +291,15 @@ class CheerioCrawler {
         checkParamOrThrow(apifyProxySession, 'options.apifyProxySession', 'Maybe String');
         checkParamOrThrow(proxyUrls, 'options.proxyUrls', 'Maybe [String]');
         checkParamOrThrow(prepareRequestFunction, 'options.prepareRequestFunction', 'Maybe Function');
+        checkParamOrThrow(apifyProxyGroups, 'options.additionalContentTypes', 'Maybe [String]');
         // Enforce valid proxy configuration
         if (proxyUrls && !proxyUrls.length) throw new Error('Parameter "options.proxyUrls" of type Array must not be empty');
         if (useApifyProxy && proxyUrls) throw new Error('Cannot combine "options.useApifyProxy" with "options.proxyUrls"!');
 
+        // Check content types
+        this.supportedContentTypes = DEFAULT_CONTENT_TYPES;
```

Comment:
There are 2 main problems: 

1) This way `this.supportedContentTypes` will point to `DEFAULT_CONTENT_TYPES` and you push the new values into `DEFAULT_CONTENT_TYPES` so if you create two cheerio crawlers they will share this settings ...

2) User can't disable `application/xml` as it's always pushed into the array

Solution:

First of, I would call that settings `opts.allowedMimeTypes`. And have this const:

```
const REQUIRED_MIME_TYPE = 'text/html';
```

then you add `allowedMimeTypes: ['application/xml', 'application/xhtml+xml']` to `DEFAULT_OPTIONS` and update for user past option to require it to be an array (as it has default value):

```
checkParamOrThrow(allowedMimeTypes, 'options.allowedMimeTypes', '[String]');
```

and finally:

 this.allowedMimeTypes = [].concat(allowedMimeTypes);
```

Reply:
Wouldn't a `Set` be better for this? Otherwise I agree.

Reply:
Correct - it should be Set